### ISSUE TITLE
Error in AbstractProtocol.php prevents sending via SMTP

### ISSUE THREAD

**Issue by wilfriedwolf** [laminas/laminas-mail#169]
Hi there,
Since release 1.14.2 the timeout check is inverted leading to time out errors where the SMTP connection is functional.
See:

https://github.com/laminas/laminas-mail/blob/d219bfdc2677cbf89de9d1c622185f1c5f1f6fbe/src/Protocol/AbstractProtocol.php#L293

Please correct. Shall I make a PR? Not quite shure why

```
if (! empty($data['timed_out'])) {
```

was changet to

```
if (! $data['timed_out']) {
```

that would probably do, if `$data['time_out']` is always set, else revert it to the original...

```
if ($data['timed_out']) {
```

Best Regards,

Willi

**Comment by Slamdunk** [comment]
Hi, fix is already in progress in https://github.com/laminas/laminas-mail/pull/167

**Comment by wilfriedwolf** [comment]
Well I just realize it is a duplicate of #167, should have written it already yesterday.. Thanks anyhow. Workaround is to fix the version to 2.14.1 :)
